File: packages/typescript/aws/src/stack-group.ts
import {
  BaseConfig,
  StackGroup as BaseStackGroup,
  StackGroupLoader,
} from '@awslv/cdk-organizer-core';
import { App } from 'aws-cdk-lib';

export class StackGroup<
  ConfigType extends BaseConfig = BaseConfig
> extends BaseStackGroup<App, ConfigType> {
  constructor(app: App, loader: StackGroupLoader, module: string) {
    super(app, loader, module);
  }
}

File: packages/python/cdk_organizer/aws/stack_group.py
"""AWS CDK Stack Group Class."""

from typing import TypeVar

from aws_cdk import App
from cdk_organizer.stack_group import StackGroup as BaseStackGroup
from cdk_organizer.stack_group import StackGroupLoader

CDK_CONFIG_TYPE = TypeVar('CDK_CONFIG_TYPE', covariant=True)


class StackGroup(BaseStackGroup[App, CDK_CONFIG_TYPE]):
    """Stack Group for AWS CDK."""

    def __init__(self, app: App, loader: StackGroupLoader) -> None:
        """Stack group constructor."""
        super().__init__(app, loader)
